tests/basic/podmetric_test.go

```diff
@@ -44,52 +45,21 @@ var _ = Describe("{PodMetricFunctional}", func() {
 		log.FailOnError(err, "Failed to update storage spec runtimeOpts")
 	})
 
-	Context("{PodMetricsLoggly}", func() {
+	Context("Sending PodMetrics to Loggly", func() {
 		namespacePrefix = "podmetricsloggly"
+		var meteringInterval time.Duration
+		var clusterUUID string
 
-		// shared test function for pod metric functional tests
-		sharedTestFunction := func() {
-			It("has to fetch the logs from loggly", func() {
-				interval, err := strconv.Atoi(meteringIntervalString)
-				log.FailOnError(err, "Failed to convert metering interval to integer")
-				meteringInterval := time.Duration(interval) * time.Minute
-				log.InfoD("Getting cluster ID")
-				clusterUUID, err := getClusterID()
-				log.FailOnError(err, "Failed to get cluster id data")
-
-				meteringData, err := getMeteringData(clusterUUID, meteringInterval)
-				log.FailOnError(err, "Failed to get metering data")
-
-				var initialPodHours float64
-				if len(meteringData) > 0 {
-					initialPodHours = getLatestPodHours(meteringData)
-				}
-				log.InfoD("Latest pod hours before starting app: %v", initialPodHours)
-
-				log.InfoD("Deploy applications")
-				contexts = make([]*scheduler.Context, 0)
-				for i := 0; i < Inst().GlobalScaleFactor; i++ {
-					contexts = append(contexts, ScheduleApplications(fmt.Sprintf("%s-%d", namespacePrefix, i))...)
-				}
-
-				log.InfoD("Validate applications")
-				ValidateApplications(contexts)
-
-				waitDuration := meteringInterval + 30*time.Second
-				log.InfoD("Wait %v for initial interval to go through in case the metering interval is after the callhome interval", waitDuration)
-				time.Sleep(waitDuration)
-
-				log.InfoD("Wait %v for previous pro-rated interval to go through", waitDuration)
-				time.Sleep(waitDuration)
-
-				log.InfoD("Wait %v for a latest interval to go through", waitDuration)
-				time.Sleep(waitDuration)
+		validatePodMetrics := func() {
+			Step("Wait for data to be consistent on loggly", func() {
+				waitForLoggly(meteringInterval)
+			})
 
+			Step("Check metering data is accurate", func() {
 				log.InfoD("Check metering data is accurate")
 
-				// try to get accurate metering data for 10mins
 				Eventually(func() bool {
```

Comment:
should change to use `task.DoRetryWithTimeout` for better error handling for aetos.